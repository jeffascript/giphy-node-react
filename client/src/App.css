:root {
    --dark: #131313;
    --app-green: #00e7cb;
    --app-green-darker: #4e9a91e6;
    --content-width: 900px;
}

@import url(https://fonts.googleapis.com/css?family=Lato:300);

.main-dark1 {
    background-color: var(--dark);
    color: #fff;
}

.blink_me {
    animation: blinker 1s linear infinite;
}

@keyframes blinker {
    50% {
        opacity: 0;
    }
}

body {
    margin: 0;
    font-family: 'Lato', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, sans-serif,
        'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
    text-align: left;
    background-color: #131313 !important;
    color: #fff !important;
}
