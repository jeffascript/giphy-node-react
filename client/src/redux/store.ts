import { configureStore } from '@reduxjs/toolkit';
import { persistReducer } from 'redux-persist';
import { combineReducers } from 'redux';
import storage from 'redux-persist/lib/storage';
import bookmarkHandlerReducer from './bookmark.slice';

const reducers = combineReducers({
    allBookmarkState: bookmarkHandlerReducer,
});

const persistConfig = {
    key: 'root',
    storage,
};

const persistedReducer = persistReducer(persistConfig, reducers);

const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: false,
        }),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export default store;
