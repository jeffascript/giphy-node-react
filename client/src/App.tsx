import React from 'react';
import './App.css';
import MyListIndicator from './components/Nav/MyListIndicator';
import SearchBox from './components/Search/SearchBox';
import CardList from './components/Media/CardList';
import { SearchHookContextProvider } from './ContextAPI/SearchHookContext';
import { StyledContainer, StyledWrapper } from './App.styles';

const MainWrapper: React.FC = ({ children, ...props }) => {
    return <StyledWrapper {...props}>{children}</StyledWrapper>;
};

function App() {
    return (
        <>
            <SearchHookContextProvider>
                <div className="container-fluid p-0 main-dark">
                    <MainWrapper>
                        <MyListIndicator />
                        <StyledContainer>
                            <SearchBox />
                            <CardList />
                        </StyledContainer>
                    </MainWrapper>
                </div>
            </SearchHookContextProvider>
        </>
    );
}

export default App;
