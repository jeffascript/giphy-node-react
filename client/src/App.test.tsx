/* eslint-disable no-irregular-whitespace */
import React from 'react';
import { render, screen, RenderResult, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from './redux/store';
import App from './App';

afterEach(cleanup);

const renderApp = (): RenderResult =>
    render(
        <Provider store={store}>
            <App />
        </Provider>,
    );

test('renders the App', () => {
    const { asFragment, container } = renderApp();
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Search For Your GIFS!!!')).toBeInTheDocument();
    expect(container.querySelector('.rs-badge-content')).toBeInTheDocument();
    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(container.querySelector('.show-grid')).not.toBeInTheDocument();
});

it('Matches Inline snapshot', () => {
    const { asFragment } = renderApp();

    expect(asFragment()).toMatchInlineSnapshot(`
        <DocumentFragment>
          <div
            class="container-fluid p-0 main-dark"
          >
            <div
              class="rs-container css-160iovx"
            >
              <div
                class="css-10g4ob8"
              >
                <div
                  class="css-1kvivmg"
                >
                  <h2>
                    Search For Your GIFS!!! 
                  </h2>
                  <div
                    class="css-iavuzk"
                  >
                    <div
                      class="rs-badge none rs-badge-wrapper"
                    >
                      <div>
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          style="font-size: 40px;"
                          viewBox="0 0 384 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
                          />
                        </svg>
                      </div>
                      <div
                        class="rs-badge-content"
                      >
                        0
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div />
              <div
                class="css-1xzcigu"
              >
                <div
                  class="css-3l4gaw"
                >
                  <div
                    class="rs-input-group css-1fesyhj rs-input-group-inside"
                  >
                    <input
                      class="rs-input css-1rxzna3 rs-input-lg"
                      placeholder="Start typing ... e.g: happy"
                      type="text"
                      value=""
                    />
                    <a
                      class="rs-btn rs-btn-default rs-input-group-btn"
                    >
                      <i
                        class="rs-icon rs-icon-search"
                      />
                      <span
                        class="rs-ripple-pond"
                      >
                        <span
                          class="rs-ripple"
                        />
                      </span>
                    </a>
                  </div>
                  <div
                    class="d-flex text-center justify-content-center align-items-center m-auto"
                    style="cursor: pointer;"
                  >
                    <svg
                      fill="currentColor"
                      height="22"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="22"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 12.6l.7.7 1.6-1.6 1.6 1.6.8-.7L13 11l1.7-1.6-.8-.8-1.6 1.7-1.6-1.7-.7.8 1.6 1.6-1.6 1.6zM1 4h14V3H1v1zm0 3h14V6H1v1zm8 2.5V9H1v1h8v-.5zM9 13v-1H1v1h8z"
                      />
                    </svg>
                     Â  
                    <span>
                      reset
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </DocumentFragment>
    `);
});
