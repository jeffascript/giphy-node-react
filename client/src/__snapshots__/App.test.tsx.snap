// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App 1`] = `
<DocumentFragment>
  <div
    class="container-fluid p-0 main-dark"
  >
    <div
      class="rs-container css-160iovx"
    >
      <div
        class="css-10g4ob8"
      >
        <div
          class="css-1kvivmg"
        >
          <h2>
            Search For Your GIFS!!! 
          </h2>
          <div
            class="css-iavuzk"
          >
            <div
              class="rs-badge none rs-badge-wrapper"
            >
              <div>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="font-size: 40px;"
                  viewBox="0 0 384 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
                  />
                </svg>
              </div>
              <div
                class="rs-badge-content"
              >
                0
              </div>
            </div>
          </div>
        </div>
      </div>
      <div />
      <div
        class="css-1xzcigu"
      >
        <div
          class="css-3l4gaw"
        >
          <div
            class="rs-input-group css-1fesyhj rs-input-group-inside"
          >
            <input
              class="rs-input css-1rxzna3 rs-input-lg"
              placeholder="Start typing ... e.g: happy"
              type="text"
              value=""
            />
            <a
              class="rs-btn rs-btn-default rs-input-group-btn"
            >
              <i
                class="rs-icon rs-icon-search"
              />
              <span
                class="rs-ripple-pond"
              >
                <span
                  class="rs-ripple"
                />
              </span>
            </a>
          </div>
          <div
            class="d-flex text-center justify-content-center align-items-center m-auto"
            style="cursor: pointer;"
          >
            <svg
              fill="currentColor"
              height="22"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 12.6l.7.7 1.6-1.6 1.6 1.6.8-.7L13 11l1.7-1.6-.8-.8-1.6 1.7-1.6-1.7-.7.8 1.6 1.6-1.6 1.6zM1 4h14V3H1v1zm0 3h14V6H1v1zm8 2.5V9H1v1h8v-.5zM9 13v-1H1v1h8z"
              />
            </svg>
             Â  
            <span>
              reset
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
